tests/engines/test_gradient_accumulation.py

```
@@ -0,0 +1,343 @@
+    # ---- suppression logic ----
+
+    @parameterized.expand(SUPPRESSION_CASES)
+    def test_suppression(self, attr_name, acc, epoch_length, num_iters, expected) -> None:
+        grad_accum = GradientAccumulation(accumulation_steps=acc)
+        original = MagicMock(name=attr_name)
+        engine = _make_engine(epoch_length)
+        setattr(engine.optimizer, attr_name, original)
+
+        saw_original: list[bool] = []
+
+        def fake_iteration(eng, batch):
```

Ruff (0.15.2)

[warning] Unused function argument: `batch`

(ARG001)